src/tests/modify.test.js

```diff
@@ -0,0 +1,661 @@
+        inputType: 'text',
+      },
+      type: 'string',
+    },
+    pet_age: {
+      title: "Pet's age in months",
+      maximum: 24,
+      'x-jsf-presentation': {
+        inputType: 'number',
+      },
+      'x-jsf-errorMessage': {
+        maximum: 'Your pet cannot be older than 24 months.',
+      },
+      type: 'integer',
+    },
+    pet_fat: {
+      title: 'Pet fat percentage',
+      'x-jsf-presentation': {
+        inputType: 'number',
+        percentage: true,
+      },
+      type: 'integer',
+    },
+    pet_address: {
+      properties: {
+        street: {
+          title: 'Street',
+        },
+      },
+    },
+  },
+  required: ['has_pet'],
+  'x-jsf-order': ['has_pet', 'pet_name'],
+  allOf: [
+    {
+      id: 'pet_conditional_id',
+      if: {
+        properties: {
+          has_pet: {
+            const: 'yes',
+          },
+        },
+        required: ['has_pet'],
+      },
+      then: {
+        required: ['pet_name', 'pet_age', 'pet_fat'],
+      },
+      else: {
+        properties: {
+          pet_name: false,
+          pet_age: false,
+          pet_fat: false,
+        },
+      },
+    },
+  ],
+};
+
+const schemaAddress = {
+  properties: {
+    address: {
+      properties: {
+        street: {
+          title: 'Street',
+        },
+        number: {
+          title: 'Number',
+        },
+        city: {
+          title: 'City',
+        },
+      },
+    },
+  },
+};
+
+const imagineSomeBasicSchema = {
+  /* ... */
+};
+
+/*
+modify() Options:
+
+{
+  fields,
+  allFields,
+  order,
+  conditionals,
+  pick,
+  omit,
+  split, 
+}
+
+*/
+
+describe('modify() - attributes', () => {
+  it('replace base field', () => {
+    const result = modify(schemaPet, {
+      fields: {
+        pet_name: {
+          title: 'Your pet name',
+        },
+        // Q:❓[*1] Do you find it useful to be a callback function instead?
+        // You can access any raw attribute from the field to do whatever verification you need,
+        // but remember to be cautious, as the attrs value might change.
+        has_pet: (fieldAttrs) => {
+          const options = fieldAttrs.oneOf?.map(({ title }) => title).join(' or ') || '';
+          return {
+            title: 'Pet owner',
+            description: `Do you own a pet? ${options}?`, // "Do you own a pet? Yes or No?"
+          };
+        },
+      },
+    });
+
+    expect(result).toMatchObject({
+      properties: {
+        pet_name: {
+          title: 'Your pet name',
+        },
+        has_pet: {
+          title: 'Pet owner',
+          description: 'Do you own a pet? Yes or No?',
+        },
+      },
+    });
+  });
+
+  it('replace nested field', () => {
+    const result = modify(schemaAddress, {
+      fields: {
+        'address.street': {
+          title: 'Street name',
+        },
+        // Q:❓ or manual nesting — both would work.
+        address: {
+          properties: {
+            number: { title: 'Door number' },
```

Comment:
In my mind, it would look like this:

```js
address: (fieldAttrs) => {
  console.log(fieldAttrs.properties.number); // { title, description, etc.. }
  return {
    properties: {
      number: { title: 'Door number' },
    },
  };
},
````

or by field path, a nested field at the time:

```js
{
  'address.street': (fieldAttrs) => ({
    title: 'Street name',
  }),
  'address.door': (fieldAttrs) => ({
    title: 'Door name',
  })
}
```

I'll add tests, making sure this is possible.